pkg/component/controller/kuberouter.go

```diff
@@ -70,6 +71,27 @@ func (k *KubeRouter) Init(_ context.Context) error { return nil }
 // Stop no-op as nothing running
 func (k *KubeRouter) Stop() error { return nil }
 
+func getHairpinConfig(cfg *kubeRouterConfig, krc *v1beta1.KubeRouter) {
```

Comment:
Maybe a better name this `applyHairpinConfig`, since this is not idempotently getting something, but updating the cfg in place.